docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -50,7 +50,15 @@ By default, the name of the decorated function, `my_asset`, is used as the asset
 
 ### Assets with dependencies
 
-Software-defined assets can depend on other software-defined assets. The easiest way to define an asset dependency is to include an upstream asset name as an argument to the decorated function. In the following example, `downstream_asset` depends on `upstream_asset`. That means that the contents of `upstream_asset` are provided to the function that computes the contents of `downstream_asset`.
+- [Basic dependencies](#defining-basic-dependencies)
+- [Explicit dependencies](#defining-explicit-dependencies)
+- [External asset dependencies](#defining-external-asset-dependencies)
+
+#### Defining basic dependencies
+
+Software-defined assets can depend on other software-defined assets. The easiest way to define an asset dependency is to include an upstream asset name as an argument to the decorated function.
+
+In the following example, `downstream_asset` depends on `upstream_asset`. That means that the contents of `upstream_asset` are provided to the function that computes the contents of `downstream_asset`.
```

Comment:
In the past, I've seen users do something like this:
```
my_asset = SourceAsset(asset_key="foo")

@asset 
def downstream_asset(my_asset)
```
Which will break because dagster will infer the upstream asset to be `AssetKey('my_asset') which won't exist. I think we should clarify in the docs that the inference is that the parameter name is the asset key to make this more explicit to users

Reply:
So to not break things, this would be:

```
my_asset = SourceAsset(asset_key="foo")

@asset 
def downstream_asset(foo)
```

Correct?

Reply:
exactly